### ISSUE TITLE
Auto-partitioning build-in feature

### ISSUE THREAD

**Issue by muntdan** [hydradatabase/hydra#62]
Hi,

Timescaledb has the built-in auto partitioning on 1 or 2 dimension (time and tenant/deviceid/etc).

Your documentation states Partitioning to be used for time partitioning, but still it can be done only manually.
It would be great if you implement something for auto-creating the partitions based on configuration of intervals and/or secondary criteria. This can help a lot on the usecase of GDPR cleaning where dropping older data would be a lot faster/easier.

**Comment by muntdan** [comment]
Based on the fact that query as is on Hydra `[cannot cope with multiple query in timely fashion](https://github.com/hydradatabase/hydra/issues/79)` and that timescaledb released the version [2.11](https://github.com/timescale/timescaledb/releases/tag/2.11.0) with support for update/delete on compressed data, I have started comparing the 2 of them again from point of view of reading the data.
My results are as fallows on same volume of data:
Hydra: 1 Month Avg single query = 5 - 5.5 seconds / 7-8cores (25-32% from 24 core) / 2-8% IOPS 
Timescaledb 1 Month Avg single query = 1 - 1.5 seconds / 5-10cores (20-35% from 24 core) / 2-8% IOPS   
with chunk_time_interval & compression set for 1 hour.

The reason I am writing this here on this issue is that I believe there is another config on my timescaledb that helps a lot for such good performance: [space partitioning](https://docs.timescale.com/api/latest/hypertable/add_dimension/) of which I do use on the above setup having configured 10 tenants that divide the customers on each table into 10 slices, so the actual query of a given client would hit only 1/10 of total chunks in order to run.

In conclusion I believe auto-partitioning feature requested here might also help a lot on the actual query performance from related [issue](https://github.com/hydradatabase/hydra/issues/79).

**Comment by wuputah** [comment]
In the near term, the [native Postgres partitioning](https://www.postgresql.org/docs/current/ddl-partitioning.html) is the way to do this for Hydra. We don't plan to add this as a primitive to the storage layer for quite some time. We have `pg_partman` on the hosted product but it is really only needed if you want to do "auto time partitioning" - it helps create new partitions for every slice of time you define (e.g. week or month). 

For "partition this into X slices by ID" you can use the "partition by hash" function. It's just a matter of writing the SQL, which is a bit repetitive but fortunately need only be done once. Here's an example:

```
CREATE TABLE customers (id INTEGER, status TEXT, arr NUMERIC) PARTITION BY HASH(id) USING columnar;
CREATE TABLE cust_part1 PARTITION OF customers FOR VALUES WITH (modulus 3, remainder 0) USING columnar;
CREATE TABLE cust_part2 PARTITION OF customers FOR VALUES WITH (modulus 3, remainder 1) USING columnar;
CREATE TABLE cust_part3 PARTITION OF customers FOR VALUES WITH (modulus 3, remainder 2) USING columnar;
```

**Comment by muntdan** [comment]
So if I define the 10 tenant manual partition on a columnar table and than insert the data via ETL every ~ 5 minutes in the main table, will the columnar stripe/chunks fallow the rules and the actual split the batch data into 10 chunks behind it ?
Also will the columnar query on the main table with where condition on the tenant will know which chunks to read ?
